src/cn-batch-forms.service.js

```diff
@@ -157,6 +162,24 @@
       }
     }
 
+    function processDiff(schema) {
+      const updateSchema = schema.params.updateSchema;
+      const links = _.filter(schema.batchConfig.links, ls => _.includes(ls, updateSchema));
+      const hardLinks = _.filter(schema.batchConfig.hardLinks, ls => _.includes(ls, updateSchema));
+      processSchemaDiff.call(this, schema.diff.schema, _.flatten(links.concat(hardLinks)));
+    }
+
+    function processSchemaDiff(properties, links) {
+      const props = _.keys(properties);
+      _.forEach(props, (prop) => {
+        if (_.has(properties[prop], "properties")) {
```

Comment:
This handles objects but I don't see a case for arrays. If a schema type is `object` it'll contain the `properties` prop, but if it's `array` it'll contain an `items` prop that contains the array child's schema